pkg/controller/containersource/reconcile.go

```diff
@@ -111,9 +111,10 @@ func (r *reconciler) Reconcile(ctx context.Context, object runtime.Object) (runt
 	expected := resources.MakeDeployment(nil, args)
 	if !equality.Semantic.DeepEqual(deploy.Spec, expected.Spec) {
 		deploy.Spec = expected.Spec
-		if r.client.Update(ctx, deploy); err != nil {
-			return object, err
-		}
+		err := r.client.Update(ctx, deploy)
+		source.Status.MarkDeploying("DeployUpdated", "Updated deployment %s", deploy.Name)
+		// Return after this update and reconcile again
+		return object, err
```

Comment:
Good catch! I agree this should return, since we need to wait for the Deployment status to update anyway.